java/kotlin-extractor/src/main/kotlin/KotlinUsesExtractor.kt

```diff
@@ -1506,13 +1533,18 @@ open class KotlinUsesExtractor(
      */
     fun getValueParameterLabel(vp: IrValueParameter, parent: Label<out DbCallable>?): String {
         val declarationParent = vp.parent
-        val parentId = parent ?: useDeclarationParent(declarationParent, false)
+        val overriddenParentAttributes = (declarationParent as? IrFunction)?.let {
```

Comment:
If we kept the declaration stack in the file extractor, we could use the below:
```
val overriddenParentAttributes = if (this is KotlinFileExtractor && this.filePath == vp.file.path && declarationParent is IrFunction) {
  this.declarationStack.findOverriddenAttributes(declarationParent)
} else {
  null
}
```
I find this more explicit in what our expectations are: the overridden attributes should only be used if we're inside the extraction of the corresponding function declaration.

Reply:
Let me know if this is a wrong assumption.

Reply:
`vp.file.path` can throw, so the correct check would be `vp.fileOrNull?.path`. or `declarationParent .fileOrNull?.path`.

I think adding the check would make it explicit that we only expect to find anything by `findOverriddenAttributes` if the files match. It would only make the intent more explicit, but it wouldn't change the behaviour. Also, performance wise having the check or not probably doesn't matter, because both `fileOrNull` and `findOverriddenAttributes` walks a parent-child declaration list linearly.

Reply:
ok done

Reply:
The same should be done in `getTypeParameterParentLabel` just to keep the symmetry between the two cases.

Reply:
Not done, because this is the case where we do need to look for type-parameter substitution happening up the stack, even in a different file -- for example, due to generic type specialisations transitively required from a synthetic function's parameter or return types.

Reply:
Makes sense. I always forget about this special case.